tensorboard/data/server/server.rs

```diff
@@ -350,6 +369,44 @@ mod tests {
         );
     }
 
+    #[tokio::test]
+    async fn test_list_plugins_multiple_timeseries_same_type() {
+        let commit = CommitBuilder::new()
+            .scalars("train", "xent2", |b| b.build())
+            .scalars("train", "xent", |b| b.build())
+            .build();
+        let handler = sample_handler(commit);
+        let req = Request::new(data::ListPluginsRequest {
+            experiment_id: "123".to_string(),
+        });
+        let res = handler.list_plugins(req).await.unwrap().into_inner();
+        assert_eq!(
+            res.plugins.into_iter().map(|p| p.name).collect::<Vec<_>>(),
+            vec!["scalars"]
+        );
+    }
+
+    // TODO(@bileschi): Add a test for the case of multiple different types of
+    // tags once sample CommitBuilders exist for plugins beyond "scalar".
```

Comment:
You can do this now. The `ScalarTimeSeriesBuilder` lets you set the
metadata of the time series, so you can construct a `SummaryMetadata`
with a different plugin name. See sibling methods for examples of
calling methods on the builder, but it should look something like:

```rust
let mut custom_metadata = pb::SummaryMetadata::default();
custom_metadata.plugin_name = "custom_scalars".to_string();
let commit = CommitBuilder::new()
    .scalars("train", "xent", |b| b.build())
    .scalars("train", "meta", |b| b.metadata(custom_metadata).build())
    .build();
```

I recommend including this test case now, because it raises a question
about the determinism of the output order, which we should address one
way or the other.